src/workerd/api/http.h

```diff
@@ -383,6 +386,9 @@ class Fetcher: public jsg::Object {
   // Wraps kj::Url::parse to take into account whether the Fetcher requires a host to be
   // specified on URLs, Fetcher-specific URL decoding options, and error handling.
 
+  jsg::Ref<Socket> connect(
+      jsg::Lock& js, kj::String address, jsg::Optional<jsg::Ref<SocketOptions>> options);
```

Comment:
Are we sure we want to add `connect()` to `Fetcher`? There is lots of code that uses `Fetcher` today that might not support `connect()` like this. For instance, the intercolo tunnel.

@kentonv @harrishancock ... thoughts?

Reply:
Happy to move it to whatever is appropriate, but for reference I saw this comment which suggested `Fetcher` was designed to grow to support more than just HTTP: https://github.com/cloudflare/workerd/blob/120ddd76781d2b828e686f176f1faa2a2a339a41/src/workerd/api/http.h#L336

Reply:
Yeah, I'm familiar with the comment, just not sure I agree with it. My preference would be a new type for this purpose.

Reply:
I'd be curious what other options we have, and how much work they'd be?

My own preference is to err on the side of greater granularity in terms of what API surface is exposed JavaScript. For example, after we implement inbound TCP connections or other new events to Workers, it'd be nice to be able to constrain the events that one worker can trigger in a worker: e.g., only "connect", but not "fetch". Presumably we could solve this attenuation problem in some other layer that causes the Fetcher to throw from the unintended methods. But having a smaller API would be more convincingly safe to me.